skrifa/src/color/color.rs

```diff
@@ -0,0 +1,259 @@
+use read_fonts::{
+    tables::colr::{ColrInstance, CompositeMode},
+    types::{BoundingBox, GlyphId, Point},
+    ReadError, TableProvider,
+};
+
+use std::{collections::HashSet, fmt::Debug};
+
+use super::{
+    colrv1_traversal::{get_clipbox_font_units, traverse_with_callbacks},
+    transforms::Transform,
+};
+
+use crate::prelude::LocationRef;
+
+#[derive(Clone, Debug, Default)]
+// This repr(C) is required so that C-side FFI's
+// are able to cast the ColorStop slice to a C-side array pointer.
+#[repr(C)]
+pub struct ColorStop {
+    pub offset: f32,
+    pub palette_index: u16,
+    pub alpha: f32,
+}
+
+// Design considerations: In principle, a local Vec allocation is not required if we're willing
```

Comment:
Good summary of the trade offs. In my mind, it comes down to how much massaging we want to do with color stops. If we’re doing a substantial amount of useful work here then I think it makes sense to pay the cost of a potential allocation (“potential” assuming we add a smallvec style optimization which is easily doable).

Reply:
Resolving, as we agreed on using a slice.

Reply:
Turns out we ran into some problems with the cast from `skrifa::color::ColorStop` to `fontations_ffi::ColorStop` (the latter needing to be a CXX type). Rolling the Skia change into the Google3 would have triggered a requirement for an `unsafe` code review whenever we touched the file with the unsafe blocks in it. So we need a wrapper type in FFI, which channels the individuals item over. I implemented that as an iterator now. So we might want to revisit making this into an iterator. 

https://skia-review.googlesource.com/c/skia/+/797756/2..3/src/ports/SkTypeface_fontations_priv.h#b91

Reply:
CC @bungeman